src/main/java/org/folio/linked/data/service/search/InstanceSearchService.java

```diff
@@ -0,0 +1,12 @@
+package org.folio.linked.data.service.search;
+
+import java.util.Collection;
+import org.folio.linked.data.domain.dto.SearchResponseTotalOnly;
+import org.springframework.http.ResponseEntity;
+
+public interface InstanceSearchService {
+
+  ResponseEntity<SearchResponseTotalOnly> searchByLccn(Collection<String> lccn);
```

Comment:
Service layer should not be aware about transport-specific things like Request/Response objects, it's violation of SRP?
Best practice is:
- Happy path:
1) Service contains only "service" logic and returns a DTO.
2) Controller wraps it in a Response object.
- Unhappy path:
1) Service throws an exception (RequestProcessingException in our case)
2) ControllerAdvice wraps it in a Error Response object.
---
Most important benefit, but not the only one: service logic is not messed with transport logic and could be re-used for different sources (Kafka for example)

Reply:
Fixed.